runtime/j9vm/jvm.c

```diff
@@ -3986,18 +3987,82 @@ JVM_LoadLibrary(const char *libName, jboolean throwOnFailure)
 			}
 #endif /* defined(J9VM_ZOS_3164_INTEROPERABILITY) */
 
-			UDATA slOpenResult = j9sl_open_shared_library((char *)libName, &handle, flags);
+			slOpenResult = j9sl_open_shared_library((char *)libName, &handle, flags);
 			Trc_SC_LoadLibrary_OpenShared(libName);
 
-/* jdk17+ calls JVM_LoadLibrary with decorated library names. If the following is done
- * then it overwrites the real error with a failure to load "liblib<name>.so.so".
- */
-#if JAVA_SPEC_VERSION < 17
 			if (0 != slOpenResult) {
-				slOpenResult = j9sl_open_shared_library((char *)libName, &handle, flags | J9PORT_SLOPEN_DECORATE);
-				Trc_SC_LoadLibrary_OpenShared_Decorate(libName);
+				char *libNameNotDecorated = (char *)libName;
+#if JAVA_SPEC_VERSION >= 17
+				/* JDK17+ jdk.internal.loader.NativeLibraries.load() calls JVM_LoadLibrary()
+				 * with a decorated library name, i.e., a path to the library name returned
+				 * from Java_java_lang_System_mapLibraryName(nameNoPrefixNoExtension).
+				 * The library name passed to j9sl_open_shared_library() with the flag
+				 * J9PORT_SLOPEN_DECORATE has to be platform independent, i.e., it must not
```

Comment:
Suggest "has to be" should be "must be".

Reply:
Updated